src/Core/Checkout/Document/Renderer/RenderedDocument.php

```diff
@@ -86,4 +96,20 @@ public function getConfig(): array
     {
         return $this->config;
     }
+
+    /**
+     * @param array<mixed> $templateOptions
+     */
+    public function setTemplateOptions(array $templateOptions): void
+    {
+        $this->templateOptions = $templateOptions;
+    }
+
+    /**
+     * @return mixed[]
```

Comment:
Having a proper options object, means you'll be able to define the return type better.